presto-cassandra/src/main/java/io/prestosql/plugin/cassandra/CassandraType.java

```diff
@@ -520,6 +474,22 @@ public Object validateClusteringKey(Object value)
         }
     }
 
+    public static boolean containsUnsupportedType(DataType dataType)
+    {
+        if (!dataType.isCollection()) {
+            Optional<CassandraType> cassandraType = toCassandraType(dataType.getName());
+            return !cassandraType.isPresent();
+        }
+
+        if (dataType.getTypeArguments().size() == 2) {
+            DataType keyType = dataType.getTypeArguments().get(0);
+            DataType valueType = dataType.getTypeArguments().get(1);
+            return containsUnsupportedType(keyType) || containsUnsupportedType(valueType);
```

Comment:
You assume it's map here. Does it have to be? Eg can it be a struct with 2 fields?
I think you should check `toCassandraType(dataType.getName()).isPresent()` here too.